### PR TITLE
enhancement: Update to use proper buttons CSS classses

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

1. Fixes buttons on groups details page, see below
2. Updates all buttons to use new css `buttton` and helper classes.
3. Removes old `btn` classses

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

Group details page before update:
image

Group details page after update:
image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Visual inspection to make sure buttons are rendered properly
2. Especially look at group details page

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated
  the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/views/groups/new.html.erb [hunk 1]
```diff
@@ -9,8 +9,11 @@
         <%= render partial: "form_fields", locals: { form: form, group: @new_group } %>
 
         <div class="my-2">
-          <%= form.submit t(:"groups.create.submit"), class: "btn btn-primary mr-1" %>
-          <%= link_to t(:"groups.create.cancel"), groups_path, class: "btn btn-default" %>
+          <%= form.submit t(:"groups.create.submit"),
+                          class: " button button--state-primary button--size-default mr-1" %>
+          <%= link_to t(:"groups.create.cancel"),
+                      groups_path,
+                      class: "button button--state-default button--size-default" %>
```
Comment (ksierks) [inline app/views/groups/new.html.erb:16]: Same as previous comment. One of the buttons is full width.
image

File: app/views/groups/new_subgroup.html.erb [hunk 2]
```diff
@@ -13,11 +13,11 @@
...
         } %>
 
-            <div class="my-2">
-              <%= form.submit t(:'groups.new_subgroup.submit'), class: "btn btn-primary mr-1" %>
-              <%= link_to t(:'groups.new_subgroup.cancel'), group_path(@group), class: "btn btn-default" %>
-            </div>
-        <% end %>
+        <div class="my-2">
+          <%= form.submit t(:'groups.new_subgroup.submit'), class: "button button--state-primary button--size-default mr-1" %>
+          <%= link_to t(:'groups.new_subgroup.cancel'), group_path(@group), class: "button button--state-default button--size-default w-full" %>
```
Comment (ksierks) [inline app/views/groups/new_subgroup.html.erb:18]: Same as previous comment. One of the buttons is full width.
image

File: app/views/projects/samples/new.html.erb [hunk 3]
```diff
@@ -23,10 +23,10 @@
 
           <div>
             <%= form.submit t("projects.samples.new.submit_button"),
-                        class: "btn btn-primary mr-1" %>
+                            class: "button button--state-primary button--size-default mr-1" %>
             <%= link_to t("projects.samples.new.cancel_button"),
-            namespace_project_samples_path,
-            class: "btn btn-default" %>
+                        namespace_project_samples_path,
+                        class: "button button--state-default button--size-default" %>
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/views/projects/samples/new.html.erb]: Thanks, that was strange.  Updated in bd98356

File: app/views/groups/new.html.erb (hunk 1 above)
Comment (joshsadam) [reply app/views/groups/new.html.erb]: Updated in bd98356

**Review by ericenns — CHANGES_REQUESTED** [review]
I can't leave a comment on the file, but the groups show page is exhibiting weird button behaviour when browser width is (1280 px).

MicrosoftTeams-image (5)

I wonder if the issue has to due with putting `flex` on the button, maybe that should be reserved to a surrounding div?

**Comment by joshsadam** [comment]
> I can't leave a comment on the file, but the groups show page is exhibiting weird button behaviour when browser width is (1280 px).
> 
> MicrosoftTeams-image (5)
> 
> I wonder if the issue has to due with putting `flex` on the button, maybe that should be reserved to a surrounding div?

Thanks @ericenns that should be fixed in 371bd64

**Review by ksierks — APPROVED** [review]
Thanks for fixing this.

**Review by ericenns — APPROVED** [review]
This looks good to me!

**Pull request merged**